Tests/SymbolKitTests/UnifiedGraph/GraphCollectorTests.swift

```diff
@@ -94,4 +94,62 @@ class GraphCollectorTests: XCTestCase {
+                                      mixins: [:])
+                            ],
+                            relationships: [])
+        
+        let (name, isMain) = GraphCollector.moduleNameFor(a, at: .init(fileURLWithPath: "A.symbols.json"))
+        XCTAssertTrue(isMain)
+        XCTAssertEqual("A", name)
+        
+        // aligned with the name "A@B.symbols.json"
+        let a_At_B = SymbolGraph(metadata: .init(formatVersion: .init(major: 1, minor: 0, patch: 0), generator: "unit-test"),
+                                 module: .init(name: "A", platform: .init()),
+                                 symbols: [
+                                     .init(identifier: .init(precise: "s:BBAatB", interfaceLanguage: "swift"),
+                                           names: .init(title: "AatB", navigator: nil, subHeading: nil, prose: nil),
+                                           pathComponents: ["B", "AatB"],
+                                           docComment: nil,
+                                           accessLevel: .init(rawValue: "public"),
+                                           kind: .init(parsedIdentifier: .class, displayName: "Class"),
+                                           mixins: [:])
+                                 ],
+                                 relationships: [])
+        
+        let (extensionName, extensionIsMain) = GraphCollector.moduleNameFor(a_At_B, at: .init(fileURLWithPath: "A@B.symbols.json"))
+        XCTAssertFalse(extensionIsMain)
+        XCTAssertEqual("B", extensionName)
+
+        // "A-snippets.symbols.json"
+        let a_snippet = SymbolGraph(metadata: .init(formatVersion: .init(major: 0, minor: 0, patch: 1), generator: "snippet-extract-unit-test-example"),
+                                    module: .init(name: "A", platform: .init(), isVirtual: true),
+                                    symbols: [
+                                        .init(identifier: .init(precise: "Snippet__A.example", interfaceLanguage: "swift"),
+                                              names: .init(title: "example", navigator: nil, subHeading: nil, prose: nil),
+                                              pathComponents: ["A", "exmaple"],
+                                              docComment: nil,
+                                              accessLevel: .init(rawValue: "public"),
+                                              kind: .init(parsedIdentifier: .snippet, displayName: "example"),
+                                              mixins: [:],
+                                              isVirtual: true)
+                                        
+                                    ],
+                                    relationships: [])
+        
+        let (snippetName, snippetIsMain) = GraphCollector.moduleNameFor(a_snippet, at: .init(fileURLWithPath: "A-snippets.symbols.json"))
+        XCTAssertFalse(snippetIsMain)
+        XCTAssertEqual("A-snippets", snippetName)
```

Comment:
Honestly, I didn't pay a huge amount of attention to the name, as the name of the module returned didn't play into the issue I was debugging, the isMain Boolean value is what reflects the issue. Because of that, I didn't touch anything in the name logic (or see what it would impact), only tweaking the isMain logic.

The test, when I created it, was based on what snippet-extractor actually dumps (or a short-form version of it), and explicitly covered what the name returns, just to get that explicitly tested. I wasn't sure of the intention of that logic, and there weren't any tests working that internal bit of name tweaking to know how (or if) I should update it.

For the specific exposure through GraphCollector, the graphs aren't indexed by name  - but the name is sourced from the graph structure itself, and not the name of the URL of the file that houses the symbolGraph.

Reply:
I did just double check that this change won't break things (yet to explain why) - I did the `export SWIFTCI_USE_LOCAL_DEPS=true` and ran through my test scenario, which is previewing the docs consistently and including all the snippets in the flow, so at least I'm confident that the change does fix the issue and does not break the code entirely.

Reply:
Digging back through all the Callers (in Docc using SymbolKit at least) of `moduleNameFor(_:at:)`. The callers are
- GraphCollect.mergeSymbolGraph(_:at:forceLoading:)
- SymbolGraphLoader.loadAll()
- SymbolGraphLoader.loadSymbolGraph(at:)
- UnifiedSymbolGraph.init(fromSingleGraph:at:)
- UnifiedSymbolGraph.merge(graph:at:)

The two UnifiedSymbolGraph discard the returned name entirely, only working with “is this the main graph” information.

The GraphCollector.mergeSymbolGraph ([GraphCollector.swift](https://github.com/swiftlang/swift-docc-symbolkit/blob/96bce1cfad4f4d7e265c1eb46729ebf8a7695f4b/Sources/SymbolKit/UnifiedSymbolGraph/GraphCollector.swift#L62)) captures the name as “extendedModuleName” and then only uses that as the “moduleName” based on the extensionGraphAssociationStrategy (only when it’s .extendedGraph). The alternative takes the name from the inputGraph internal data (inputGraph.module.name). 
- Digging this back, the extensionGraphAssociation defaults to .extendedGraph

The SymbolGraph loader ([SymbolGraphLoader.swift](https://github.com/swiftlang/swift-docc/blob/93a39a698cd74781df7a3b7cf0a7cd298ee02fbe/Sources/SwiftDocC/Infrastructure/Symbol%20Graph/SymbolGraphLoader.swift#L56)) `loadAll()` uses the name to invoke addDefaultAvailability, and if it’s not a “main” module, to invoke `ExtendedTypeFormatTransformation.transformExtensionBlockFormatToExtendedTypeFormat(&symbolGraph, moduleName: moduleName)`, but otherwise doesn’t use the name - as it tends to reference the symbol graphs by URL.

The other place SymbolGraph loader uses is ([loadSymbolGraph(at:)](https://github.com/swiftlang/swift-docc/blob/93a39a698cd74781df7a3b7cf0a7cd298ee02fbe/Sources/SwiftDocC/Infrastructure/Symbol%20Graph/SymbolGraphLoader.swift#L191)) captures the name, and changes the name IN the symbol graph only if the graph isn’t a main module and has no bystanders.

Based on all this, I think the name happens to be slipping through the cracks, but your sense of “hey, that’s likely wrong” is correct in that it could easily break some assumptions elsewhere in the code. So while it doesn’t break anything now, it’s not “correct” as it stands for snippets.

…

The logic in [moduleNameFor](https://github.com/swiftlang/swift-docc-symbolkit/blob/96bce1cfad4f4d7e265c1eb46729ebf8a7695f4b/Sources/SymbolKit/UnifiedSymbolGraph/GraphCollector.swift#L128) uses returns the name for the module from the symbol graph itself (graph.module.name), and only futzes with the name of the URL in the case of an extension. What I’m uncertain of is what the *correct* logic is here. 

My sample code has an extension onto some core Swift types, and as I looked through the symbol graph, it’s `module.name` _was_ correct, so I’m unclear on when it isn’t correct and should be pulled form the name of the URL. The parsing code references cross-import overlays, with an example name that shows cascading frameworks, and also assumes that the name will always be in the form Framework1@Framework2. The snippets extractor writes out the name of the graph as “Framework1-snippets.symbols.json”.  Based on that, I could put in some additional logic to check the form and go from there. 

What do you think of a special case looking for “-snippets” in that name, and in that case returning the value from the graph itself (graph.module.name?)

Reply:
(added my guess-work "fix" as a commit) - let me know what you think, and if there's a different way you'd prefer to approach this